app/src/routes/index.tsx

```diff
@@ -34,92 +44,102 @@ const UserRoute = () => {
 };
 
 const PermissionsRoute = (props: any) => {
-  const fbContext = useContext(FirebaseContext);
-
-  if (props?.validator && props.validator(fbContext?.role)) {
+  const { currentUser } = useAuth();
+  if (props?.validator && props.validator(currentUser?.role)) {
```

Comment:
how come y'all are using typescript but then not using types? 😛  haha if you define interfaces for your props you can skip the `if props.validator` check